src/attr.rs

```diff
@@ -24,12 +26,55 @@ pub fn valid<I: Iterator<Item = char>>(chars: I) -> (usize, bool) {
     (p.pos, has_attr)
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Attr<'s> {
+    raw: CowStr<'s>,
+}
+
+impl<'s> Attr<'s> {
+    pub fn write<F: FnMut(&str) -> fmt::Result>(&self, mut write_fn: F) -> fmt::Result {
```

Comment:
Is there an advantage of this method over returning a &str iterator? A &str iterator seems more general? Which is useful when it is part of the public API.

We should be able to do simply
```rs
attr.parts().try_for_each(|p| self.write_escape(p))?;
```
in html.rs then.

Reply:
The limitation mainly comes from needing more mutable state than the AttrChars iterator. This comes from e.g. the escape state toggle, no backslashes added to the first part, etc. I could of course add these to an iterator struct, but that would add more code bloat, and I simply thought this way would be simpler.

If you prefer iterators, then I have no issue with changing to it :smile:

Reply:
Yeah, in my opinion, iterators seem more convenient from the client pov, otherwise you are tied to the str -> format::error function signature. You might not always be interested in writing, could be reading only for example. I think the performance should be more or less the same between the two different methods.

I guess we are not forced to have a dedicated iterator struct, could just use std::iter::from_fn and return an impl Iterator<&str>. But a struct may be more user-friendly because it can be named in type signatures, I suppose.

Reply:
Got it! Will add ASAP

Reply:
Implemented now, with the `AttributeValueParts` iterator.